tests/arc/texceptions.nim

```diff
@@ -14,3 +14,13 @@ block: # issue #13071
       a = e.msg & $e.name # was segfaulting here for `nim cpp --gc:arc`
     doAssert a == "foo:MyExcept"
   fun()
+
+
+block: # issue #17351
+  type
+    Foo {.inheritable.} = object
```

Comment:
wrong file (`texceptions.nim`): this isn't related to exceptions nor to arc; instead put it in for eg:
`tests/misc/t17351.nim`
with:
```nim
discard """
  targets: "c cpp"
"""
```
(and optionally with `matrix "--gc:arc; --gc:refc"`)